packages/world/src/requireNamespace.sol

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.21;
+
+import { ResourceId, WorldResourceIdInstance } from "./WorldResourceId.sol";
+import { ROOT_NAME } from "./constants.sol";
+import { RESOURCE_NAMESPACE } from "./worldResourceTypes.sol";
+import { IWorldErrors } from "./IWorldErrors.sol";
+
+using WorldResourceIdInstance for ResourceId;
+
+/**
+ * @notice Checks if a given `resourceId` is a namespace.
+ * @dev Reverts with IWorldErrors.World_InvalidResourceType if the ID does not have the correct components.
+ * @param resourceId The resource ID to verify.
+ */
+function requireNamespace(ResourceId resourceId) pure {
+  if (resourceId.getType() != RESOURCE_NAMESPACE || resourceId.getName() != ROOT_NAME) {
```

Comment:
just realized another way we could do this is

```
if (resourceId.unwrap() == resourceId.getNamespaceId().unwrap()) {
```

Reply:
This suggestion is actually more gas efficient. I like it, this functionality is encapsulated anyway